From fa93bfcc93a598b0a50de42f8eceddab150d81b5 Mon Sep 17 00:00:00 2001
From: Patrick Combes <patrick.combes@intel.com>
Date: Tue, 30 Apr 2013 09:16:48 +0200
Subject: Fix possible uninitialized variable

BZ: 104381

KW reports possible unitialized variable with usage of 'p_name'
'tmp_path' should be preferred over 'p_name' to avoid this unitialized
pointer condition.

Category: aosp improvement
Domain: CWS.BT-Core
Origin: internal
Upstream-Candidate: yes

Change-Id: Ibcb93c00e79f6e5552fff9ea441f8736c51cd8ca
Orig-Change-Id: Ie40c86dc9d430635e9f4690fedc75fc59934b8a7
Signed-off-by: Patrick Combes <patrick.combes@intel.com>
Reviewed-on: http://android.intel.com:8080/105503
Reviewed-by: cactus <cactus@intel.com>
Reviewed-by: Gros, Julien <julien.gros@intel.com>
Reviewed-by: Yao, Yong <yong.yao@intel.com>
Tested-by: Champciaux, Nicolas <nicolas.champciaux@intel.com>
Reviewed-by: buildbot <buildbot@intel.com>
Tested-by: buildbot <buildbot@intel.com>
Reviewed-on: http://android.intel.com:8080/123805
Reviewed-by: Poignet, Cedric <cedric.poignet@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 src/hardware.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/hardware.c b/src/hardware.c
index 37f9970..ce6cf21 100644
--- a/src/hardware.c
+++ b/src/hardware.c
@@ -745,7 +745,7 @@ check_local_name:
                     {
                         if ((hw_cfg_cb.fw_fd = open(tmp_path, O_RDONLY)) == -1)
                         {
-                            ALOGE("vendor lib preload failed to open [%s]", p_name);
+                            ALOGE("vendor lib preload failed to open [%s]", tmp_path);
                         }
                         else
                         {
-- 
1.7.4.1

